amaranth_boards/resources/interface.py

```diff
@@ -142,3 +142,20 @@ def PS2Resource(*args, clk, dat, conn=None, attrs=None):
         ios.append(attrs)
 
     return Resource.family(*args, default_name="ps2", ios=ios)
+
+
+def RGMIIResource(*args, txc, txd, tx_ctl, rxc, rxd, rx_ctl, mdc, mdio, attrs=None, conn=None):
```

Comment:
- Should `mdc` and `mdio` be optional pins?
- Should there be an optional `rst_n` pin?